--- origsrc/lua-5.1.5/etc/lua.pc	2012-02-10 03:50:30.000000000 -0600
+++ src/lua-5.1.5/etc/lua.pc	2015-08-23 11:42:57.347961200 -0500
@@ -8,24 +8,24 @@ V= 5.1
 R= 5.1.5
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /usr
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
-INSTALL_MAN= ${prefix}/man/man1
+INSTALL_MAN= ${prefix}/share/man/man1
 INSTALL_LMOD= ${prefix}/share/lua/${V}
 INSTALL_CMOD= ${prefix}/lib/lua/${V}
 
 # canonical vars
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+includedir=${prefix}/include/lua-${V}
 
 Name: Lua
 Description: An Extensible Extension Language
 Version: ${R}
 Requires: 
-Libs: -L${libdir} -llua -lm
+Libs: -L${libdir} -llua-${V} -lm
 Cflags: -I${includedir}
 
 # (end of lua.pc)
